internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -234,13 +236,133 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 		}
 		return r.installDay1(ctx, log, clusterDeployment, clusterInstall, cluster, pullSecret)
 	} else if swag.StringValue(cluster.Kind) == models.ClusterKindAddHostsCluster {
+		// Set HTTPS Ignition URL
+		log.Debugf("1. ACI status APIVIP: %v", clusterInstall.Status.APIVIP)
+		mcsSecretKey, err := r.createMCSCertSecret(ctx, log, clusterDeployment, cluster, clusterInstall)
+		if err != nil {
+			log.WithError(err).Error("failed to save spoke MCS certificate")
+			return r.updateStatus(ctx, log, clusterInstall, nil, err)
+		}
+		log.Debugf("2. ACI status APIVIP: %v", clusterInstall.Status.APIVIP)
+		if err := r.setHTTPSIgnitionURL(ctx, log, cluster, clusterInstall, mcsSecretKey); err != nil {
+			log.WithError(err).Error("failed to set https ignition URL")
+			return r.updateStatus(ctx, log, clusterInstall, nil, err)
+		}
+		log.Debugf("3. ACI status APIVIP: %v", clusterInstall.Status.APIVIP)
 		// Day 2
 		return r.installDay2Hosts(ctx, log, clusterDeployment, clusterInstall, cluster)
 	}
 
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) setHTTPSIgnitionURL(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster, clusterInstall *hiveext.AgentClusterInstall, secretRef types.NamespacedName) error {
+
+	if clusterInstall.Spec.IgnitionEndpoint == nil {
+		log.Debugf("Empty endpoint found")
+		// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+		// ensureMCSCert would add ignition override with MCS secret
+		ignitionEndpoint := fmt.Sprintf("%s/config/%s", common.GetMCSUrlBase(cluster), models.HostRoleWorker)
```

Comment:
should not pool part be per host? I think that mcs pool can be specified per host so we should actually set endpoint without pool and add pool part afterwards? no?